src/uct/ib/base/ib_iface.c

```diff
@@ -1322,11 +1366,18 @@ uct_ib_iface_init_roce_addr_prefix(uct_ib_iface_t *iface,
     ucs_assert(uct_ib_iface_is_roce(iface));
 
     if ((gid_info->roce_info.ver != UCT_IB_DEVICE_ROCE_V2) ||
-        !config->rocev2_local_subnet) {
+        (!config->rocev2_local_subnet &&
+        (config->reachability_mode != UCT_IB_REACHABILITY_MODE_LOCAL_SUBNET))) {
         iface->addr_prefix_bits = 0;
         return UCS_OK;
     }
 
+    /* Override the original value of reachability_mode if
+       rocev2_local_subnet is enabled to maintain backward compatibility */
+    iface->config.reachability_mode = config->rocev2_local_subnet ?
+                                      UCT_IB_REACHABILITY_MODE_LOCAL_SUBNET :
```

Comment:
maybe unite common logic with lines 1369-1370 so a single helpe function will return the reachability mode based on config, and it would consider both rocev2_local_subnet and reachability_mode fields for backward compat?